### ISSUE TITLE
CBC-MAC gives a wrong output

### ISSUE THREAD

**Issue by bricecesarin** [krzyzanowskim/CryptoSwift#746]
The initial goal is to re-write an Android code done with bouncyCastle library:

```
val macKey = "89D7B23D500D492FA01DC53B44864AB8".hexStringToByteArray()
val cipheredData = "E77A914D5C94A04B6D8E10BA7A56A015AC2C40167F867A97B6349F29F3100D6D".hexStringToByteArray()

var macBlock = CBCBlockCipherMac(AESEngine(), ISO7816d4Padding())
var output = ByteArray(8)
macBlock.init(KeyParameter(macKey))
macBlock.update(cipheredData, 0, cipheredData.size)
macBlock.doFinal(output, 0)
```

To do so, I wrote the equivalent for  CryptoSwift library: 

```
let macKey = Data(hex: "89D7B23D500D492FA01DC53B44864AB8")
let cipheredData = Data(hex: "E77A914D5C94A04B6D8E10BA7A56A015AC2C40167F867A97B6349F29F3100D6D")

do {
    let output = try CBCMAC(key: macKey.bytes).authenticate(cipheredData.bytes)
} catch {
    debugPrint("exception \(error)")
}
```

The output is not the same, it is not even the same size as the Android code which is a 8 bytes result (as given in the output). CBC-MAC should have a 8 bytes output.

**Comment by owlstead** [comment]
This is because it actually seems to implement CMAC, which is an error. CBC-MAC just uses one key, but is insecure for dynamically sized messages.

The output size *may* be cut to 8 bytes using the leftmost bytes, but the full output size is still 16 bytes, so that's not wrong.

**Comment by krzyzanowskim** [comment]
Fixed by #749
